activesupport/lib/active_support/claims.rb

```diff
@@ -0,0 +1,63 @@
+require 'active_support/time'
+
+module ActiveSupport
+  # +Claims+ supports <tt>ActiveSupport::MessageVerifier</tt> by
+  # representation of message in the form JWT Claims and verifying
+  # the claims of the signed message. 
+  class Claims
+    class InvalidClaims < StandardError; end
+    class ExpiredClaims < StandardError; end
+    
+    attr_reader :payload, :purpose, :expires_at
+
+    # Keyword argument +:payload+ is mandatory
+    def initialize(payload:, **options)
+      @payload = payload
+      @purpose = self.class.pick_purpose(options)
+      @expires_at = pick_expiration(options)
+    end
+
+    class << self
+      attr_accessor :expires_in
+
+      def pick_purpose(options)
+        options.fetch(:for) { 'universal' }
+      end
+
+      def verify!(claims, options = {})
+        raise InvalidClaims if claims[:for] != pick_purpose(options)
```

Comment:
* You can inline this and use a constant instead.
* This method is a bit hard to understand since there're two tailing ifs. Consider making them as a proper `if/else` block will make the code cleaner.